### PR TITLE
 Add support for building discovery image using other distros rocky8/alma8 in addition to centos stream 8

### PR DESCRIPTION
**Pull request by nixfu**
Have tested building the discovery image using rocky8 and alma8.  For those who want to use a distro closer to RHEL.  This change makes a few things variablized and allows the user to simply do any of the following:

./build-livecd fdi-alma8.ks
./build-livecd fdi-centos8.ks
./build-livecd fdi-rocky8.ks

### CODE REVIEW

File: build-livecd-root [hunk 1]
```diff
@@ -23,6 +23,22 @@ fi
...
+
+if [ -z ${DISTRO_INSTALL_NAME} ]; then
+	DISTRO_INSTALL_NAME="Centos 8 Stream"
+fi
+
+if [ -n ${CUSTOM_LORAX_DIR} ]; then
+	LORAX_TEMPLATES="--lorax-templates ${CUSTOM_LORAX_DIR}"
+else
+	LORAX_TEMPLATES=""
+fi
```
Comment (lzap) [inline build-livecd-root:39]: This is not correct shell code:

```
[lzap@nuc foreman-discovery-image]$ echo $CUSTOM_LORAX_DIR

[lzap@nuc foreman-discovery-image]$ [ -n ${CUSTOM_LORAX_DIR} ] && echo BAD
BAD
```

Your patch does not work when the variable is not set, then it breaks lorax as it passes the argument without any directory.

**Review by lzap — CHANGES_REQUESTED** [review]
Thanks, does not work tho :)

**Comment by nixfu** [comment]
Added fix for issue.

**Comment by nixfu** [comment]
Added fix for issue.

**Comment by lzap** [comment]
@nixfu just few more days, sorry, we are doing a maintenance handover for this project.

**Comment by lzap** [comment]
@ares hey can someone take this review? I am good but this needs testing to ensure it does not break CentOS Stream building.

**Comment by jloeser** [comment]
Rocky/Alma URLs to installation ISOs are outdated (8.5 → 8.7).

For Rocky I get the following error:
```
09:34:38,026 CRIT anaconda:anaconda: exception: Traceback (most recent call last):#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/payload/dnf/payload.py", line 541, in _apply_selections#012    self._base.install_specs(install=include_list, exclude=exclude_list)#012#012  File "/usr/lib/python3.6/site-packages/dnf/base.py", line 2011, in install_specs#012    module_depsolv_errors=module_depsolv_errors)#012#012dnf.exceptions.MarkingErrors: Problems in request:#012missing packages: rubygem-facter#012#012#012During handling of the above exception, another exception occurred:#012#012#012Traceback (most recent call last):#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/threading.py", line 280, in run#012    threading.Thread.run(self)#012#012  File "/usr/lib64/python3.6/threading.py", line 885, in run#012    self._target(*self._args, **self._kwargs)#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation.py", line 406, in run_installation#012    queue.start()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 311, in start#012    item.start()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 311, in start#012    item.start()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 311, in start#012    item.start()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 490, in start#012    self.run_task()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 456, in run_task#012    self._task(*self._task_args, **self._task_kwargs)#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/payload/dnf/payload.py", line 1372, in install#012    self.check_software_selection()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/payload/dnf/payload.py", line 1119, in check_software_selection#012    self._apply_selections()#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/payload/dnf/payload.py", line 553, in _apply_selections#012    self._payload_setup_error(e)#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/payload/dnf/payload.py", line 729, in _payload_setup_error#012    if errors.errorHandler.cb(exn) == errors.ERROR_RAISE:#012#012  File "/usr/lib64/python3.6/site-packages/pyanaconda/errors.py", line 304, in cb#012    raise NonInteractiveError("Non interactive installation failed: %s" % exn)#012#012pyanaconda.errors.NonInteractiveError: Non interactive installation failed: Problems in request:#012missing packages: rubygem-facter
```

**Comment by lzap** [comment]
@stejskalleos perhaps?

**Comment by lzap** [comment]
Leos is not available for a while, when he is back this will be reviewed.

**Comment by jloeser** [comment]
> Leos is not available for a while, when he is back this will be reviewed.

No hurry. I just wanted to check what's already there for building FDI images.

**Comment by lzap** [comment]
Hacker!!!! :-)

Ok I think tcpdump is fine. @stejskalleos you there?

**Comment by stejskalleos** [comment]
Yeah, I need to prioritize it and put it in my prio-backlog. Will take a look ASAP

**Comment by stejskalleos** [comment]
Hi,
sorry for the late review. Building image for Centos seems to be working, but not for Alma linux:
```
./build-livecd fdi-alma8.ks
Now run as root ./build-livecd-root or submit fdi-image.ks into koji

./build-livecd-root 4.2 ./result nonvirt

Error creating livecd, use KEEP_TMPDIR to investigate
==> livemedia.log <==
2023-05-25 12:58:41,151 DEBUG livemedia-creator: Namespace(anaconda_args=['--product Alama Linux 8'], app_file='appliance.xml', app_name=None, app_template=None, arch=None, armplatform=None, compress_args=[], compression='xz', disk_image=None, domacboot=False, dracut_args=['--xz', '--no-hostonly', '--debug', '--no-early-microcode', '--omit plymouth', '--add livenet dmsquash-live convertfs pollcdrom qemu qemu-net', '--add-drivers mptbase mptscsih mptspi hv_storvsc hid_hyperv hv_netvsc hv_vmbus'], extra_boot_args='nonvirt', fs_image=None, fs_label='Anaconda', image_name=None, image_only=False, image_size_align=0, image_type=None, iso='/root/foreman-discovery-image-upstream/install.iso', iso_name=None, iso_only=False, keep_image=False, kernel_args=None, ks=['/root/foreman-discovery-image-upstream/fdi-image.ks'], live_rootfs_keep_size=False, live_rootfs_size=0, location=None, logfile='/root/foreman-discovery-image-upstream/livemedia.log', lorax_templates=None, make_ami=False, make_appliance=False, make_disk=False, make_fsimage=False, make_iso=True, make_oci=False, make_ostree_live=False, make_pxe_live=False, make_tar=False, make_tar_disk=False, make_vagrant=False, no_kvm=False, no_virt=False, oci_config=None, oci_runtime=None, ovmf_path='/usr/share/edk2/ovmf/', project='FDI', proxy=None, qcow2=False, qemu_args=[], ram=2048, releasever='4.2/41bceaa', result_dir=None, squashfs_args=None, tar_disk_name=None, timeout=None, title='Linux Live Media', tmp='/var/tmp/fdi-xRRnQeScX', vagrant_metadata=None, vagrantfile=None, vcpus=2, virt_uefi=False, vnc=None, volid=None, with_rng='/dev/random')
2023-05-25 12:58:41,151 INFO livemedia-creator: livemedia-creator v28.14.70-1
2023-05-25 12:58:41,151 INFO pylorax: selinux is enabled and in Enforcing mode
2023-05-25 12:58:41,204 INFO pylorax: disk_img = /var/tmp/fdi-xRRnQeScX/lmc-disk-314ziodu.img
2023-05-25 12:58:41,205 INFO pylorax: Using disk size of 3002MiB
2023-05-25 12:58:41,205 INFO pylorax: install_log = /root/foreman-discovery-image-upstream/virt-install.log
2023-05-25 12:58:41,205 DEBUG pylorax.imgutils: make tmp mountdir /var/tmp/fdi-xRRnQeScX/lorax.imgutils.sq6pnn1l
2023-05-25 12:58:41,230 ERROR livemedia-creator: Command '['mount', '-o', 'loop', '/root/foreman-discovery-image-upstream/install.iso', '/var/tmp/fdi-xRRnQeScX/lorax.imgutils.sq6pnn1l']' returned non-zero exit status 32.
```

**Comment by lzap** [comment]
Leoš, as long as it does not break RHEL and CentOS Stream builds, feel free to merge.

**Review by stejskalleos — CHANGES_REQUESTED** [review]
Generating the image doesn't work for me. Running:
```
./build-livecd fdi-centos8.ks
./build-livecd-root 4.2 ./result nonvirt
```

`ll ./result` returns empty folder and no image is generated. Works fine on the `master` branch.

**Comment by stejskalleos** [comment]
Hi, I'm closing the PR, it's been a while since the last update and per my last comment, the iso generation doesn't work.
Feel free to re-open the issue if you think otherwise.

**Pull request closed without merging**